numpy/lib/tests/test_index_tricks.py

```diff
@@ -566,3 +566,123 @@ def test_ndindex():
     # Make sure 0-sized ndindex works correctly
     x = list(ndindex(*[0]))
     assert_equal(x, [])
+
+
+
+# ndindex test cases for reimplementation of ndindex using itertools 
+# 1.  Empty Shapes and Zero Dimensions
+def test_ndindex_empty_and_zero_dimensions():
+    # Empty shape
+    assert list(np.ndindex()) == [()]
```

Comment:
It probably does make sense to add a few more test cases, but note that the already-existing `test_ndindex` already has this particular test case for example, and may cover one or two other cases.